### PR TITLE
Timeline improvements

### LINKED ISSUE #9: Resource availability timeline
### ResourceAvailability
image
- No grouping / ordering selector.
#### Props
- `date`: Date the timeline is shown for.
- `groups`: List of groups. Group info contains name + list of resource ids.
- `onDateChange`
#### Subcomponents
```js
<div className="top-left">{/* empty, might contain ordering / grouping in the future */}</div>
<DateSelector onChange={props.onDateChange} value={props.date} />
<Sidebar groups={props.groups} />
<GroupedTimeline date={props.date} groups={props.groups} />
```
---
### DateSelector
image
#### Props
- `value`: Selected date.
- `onChange`
#### Subcomponents
```js
<div onClick={props.onChange(props.value - 1)}>edellinen</div>
<div>{props.value}</div>
<div onClick={props.onChange(props.value + 1)}>seuraava</div>
```
---
### Sidebar
image
#### Props
- `groups`: List of groups. Group info contains name + list of resource ids.
#### Subcomponents
```js
{props.groups.map(group => <GroupInfo {...group} />)}
```
---
### GroupInfo
image
#### Props
- `name`
- `resources`: List of resource ids.
#### Subcomponents
```js
<div>{props.name}</div>
{props.resources.map(resource => <ResourceInfo id={resource} />)}
```
---
### ResourceInfoContainer (connected)
image
#### Props
- `id`
#### Selector
- `name`
- `peopleCapacity`
#### Subcomponents
```js
<div>{props.name}</div>
<div>{props.capacity}</div>
```
---
### GroupedTimeline
image
#### Props
- `date`: Date of the timeline.
- `groups`: List of groups
#### Subcomponents
```js
{props.groups.map(group => <GroupTimeline date={props.date} {...group} />)}
```
---
### GroupTimeline
image
#### Props
- `date`: Date of the timeline.
- `resources`: List of resource ids.
#### Subcomponents
```js
<div>{/* hours here */}</div>
{props.resources.map(resource => <ResourceReservationsTimelineContainer date={props.date} id={resource} />)}
```
---
### ResourceReservationsTimelineContainer (connected)
image
#### Props
- `date`: Date of the timeline.
- `id`
#### Selector
- `items`: List of reservations and their data (name, begin, end, ...) and empty slots for the day and resource.
#### Subcomponents
```js
<ReservationTimeline items={props.items} />
```
---
### ReservationsTimeline
image
#### Props
- `items`: List of reservations and their data (name, begin, end, ...) and empty slots for the day and resource.
#### Subcomponents
```js
{props.items.map(item =>
  item.type === 'reservation'
  ? <Reservation {...item} />
  : <ReservationSlot />
)}
```
---
### Reservation
image
#### Props
- `name`
- `begin`
- `end`
#### Subcomponents
```js
<div>{begin} - {end}</div>
<div>{name}</div>
```
---
### ReservationSlot
image

## Other things
- Show reservation info on hover? Currently 30 minute and 60 minute reservations are impossible to see.
- `onClick` handler for ReservationSlot. Should be given originally to ResourceAvailability?
- Hour markers (borders) to span down through the group.
- Drag to scroll sideways? On desktop sideways scrolling can be quite difficult.

### PR DESCRIPTION
**Pull request by FelixLoether**
Built on top of #22. Closes #9.

### CODE REVIEW

File: app/shared/availability-view/TimelineGroups/TimelineGroup/AvailabilityTimeline/ReservationSlot.js [hunk 1]
```diff
@@ -1,7 +1,26 @@
...
+  }).isRequired,
+  resourceId: PropTypes.string.isRequired,
+};
+export default function ReservationSlot(props) {
+  const query = queryString.stringify({
+    begin: props.begin.format(),
+    reserve: true,
+  });
+  return (
+    <a
```
Comment (lraivio) [inline app/shared/availability-view/TimelineGroups/TimelineGroup/AvailabilityTimeline/ReservationSlot.js]: - [ ] This should be a react-router `Link` so the whole page isn't refreshed when the link is clicked.

**Comment by lraivio** [comment]
Looks very good!
- [ ]  I think you can squash the fixups here and do the final link change and then we can just merge this branch to make merging the timeline more streamlined.

**Comment by FelixLoether** [comment]
Changes + squashes made.

**Comment by FelixLoether** [comment]
Fixed.

**Pull request merged**